src/psyclone/tests/domain/nemo/transformations/nemo_arrayrange2loop_trans_test.py

```diff
@@ -271,29 +466,30 @@ def test_apply_different_num_dims():
     with pytest.raises(InternalError) as info:
         trans.apply(array_ref.children[1])
     assert ("The number of ranges in the arrays within this "
-            "assignment are not equal. This is invalid PSyIR and "
-            "should never happen." in str(info.value))
+            "assignment are not equal. Any such case should have "
+            "been dealt with by the validation method or represents "
+            "invalid PSyIR." in str(info.value))
 
 
 def test_apply_array_valued_function():
-    '''Check that the apply method does not modify range nodes when they are
-    used to specify the part of an array to pass into an array valued
-    function.
-
+    ''' Check that the apply the transformation refuses to transform the
+    assignment when range nodes are inside a function, as it does not know
+    if the function is declared as 'elemental' which changes the semantics
+    of the array notation.
     '''
     _, invoke_info = get_invoke("array_valued_function.f90", api=API, idx=0)
     schedule = invoke_info.schedule
     assignment = schedule[1]
     array_ref = assignment.lhs
     trans = NemoArrayRange2LoopTrans()
-    trans.apply(array_ref.children[2])
-    writer = FortranWriter()
-    result = writer(schedule)
-    assert (
-        "  jn = 2\n"
-        "  do jk = 1, jpk, 1\n"
-        "    z3d(1,:,jk) = ptr_sjk(pvtr(:,:,:),btmsk(:,:,jn) * btm30(:,:))\n"
-        "  enddo" in result)
+    with pytest.raises(TransformationError) as info:
+        trans.apply(array_ref.children[2])
+    # The PSyIR currently sees imported symbols that can be functions or arrays
```

Comment:
The issue is that fparser considers the function call expression an array accessor because it doesn't keep information about what the symbol is to disambiguate them. I can't find a specific issue in fparser although there is the generic "capture symbol information". Do you want me to reference that one? I know you where adding symbol tables, but this is a tricky case as it comes from an import.

Reply:
Yes, it's going to be hard to fix this. I think a reference to the generic issue will do.

Reply:
Done